File: src/TmrOverlay.App/Overlays/FuelCalculator/FuelCalculatorOverlayDefinition.cs
using TmrOverlay.Core.Overlays;

namespace TmrOverlay.App.Overlays.FuelCalculator;

internal static class FuelCalculatorOverlayDefinition
{
    public static OverlayDefinition Definition { get; } = new(
        Id: "fuel-calculator",
        DisplayName: "Fuel Calculator",
        DefaultWidth: 503,
        DefaultHeight: 315,
        Options:
        [
            OverlaySettingsOptionDescriptor.Boolean(
                OverlayOptionKeys.FuelAdvice,
                "Show advice column",
                defaultValue: true)
        ],
        FadeWhenLiveTelemetryUnavailable: true,
        ContextRequirement: OverlayContextRequirement.LocalPlayerInCarOrPit);
}

File: local-mac/TmrOverlayMac/Sources/TmrOverlayMac/Overlays/FuelCalculator/FuelCalculatorOverlayDefinition.swift
import AppKit

enum FuelCalculatorOverlayDefinition {
    static let definition = OverlayDefinition(
        id: "fuel-calculator",
        displayName: "Fuel Calculator",
        defaultSize: NSSize(width: 600, height: 340),
        fadeWhenLiveTelemetryUnavailable: true
    )
}
